### ISSUE TITLE
Pandoc omits the role and aria-label from body > section.footnotes

### ISSUE THREAD

**Issue by apatel762** [apatel762/backlinks-ssg#81]
Pandoc generates a `<section>` at the bottom of the page with the `.footnotes` class on the element when there are footnotes in the Markdown notes.

Google's Lighthouse then raises an accessibility issue re: the footnotes section because it doesn't contain a `role` and `aria-label` to help screen readers to understand what is in the section.

> ARIA - These are opportunities to improve the usage of ARIA in your application which may enhance the experience for users of assistive technology, like a screen reader.
> 
> Elements with an ARIA `[role]` that require children to contain a specific `[role]` are missing some or all of those required children.
> 
> Failing Elements
> * `section.footnotes`

## Using Pandoc

The fix might be to look through the different Pandoc options and see if there's something that could add the ARIA labels to the footnotes section. Even if we just use static text (i.e. "The footnotes section") to describe the section it would be fine, because it really is the same on all pages.

Maybe a Pandoc Lua filter? Not sure if you could add something to the HTML writer itself to make it add the ARIA labels, but worth a look.

## Using JavaScript

The alternative method for fixing this, which I'm not a huge fan of, would be to add some JavaScript in-line in the header which adds the ARIA `[role]` and label for us if the footnotes section is present.

The downside of this is that every single page will have JavaScript running which really shouldn't be needed given that we're just generating a site to look at some plaintext notes.

## Using Python

Could figure out a way of doing it in the static site generator itself? Not sure how it would work but if all else fails :shrug: 

---

1. WCAG Working Group (July 24, 2014). "[Using HTML5 section element](https://www.w3.org/WAI/GL/wiki/Using_HTML5_section_element#Example_2:_Structuring_HTML5_content_with_ARIA_and_the_.3Csection.3E_element)". *[Archived](https://web.archive.org/web/20201112003141/https://www.w3.org/WAI/GL/wiki/Using_HTML5_section_element)*. Retrieved May 23, 2021.
2. Google Developers (May 2, 2019). "[Elements with an ARIA [role] that require children to contain a specific [role] are missing some or all of those required children](https://web.dev/aria-required-children/)". *[Archived](https://web.archive.org/web/20210523131012/https://web.dev/aria-required-children/)*. Retrieved May 23, 2021.
